From e89054244dd38b8d3006858d6dbabe673c09e5a8 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:12:12 -0400
Subject: [ethernet] igb: Cleanups to fix assignment in if error

Message-id: <1409062359-13063-55-git-send-email-sassmann@redhat.com>
Patchwork-id: 89178
O-Subject: [RHEL7.1 PATCH 54/81] igb: Cleanups to fix assignment in if error
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Carolyn Wyborny <carolyn.wyborny@intel.com>

This patch fixes ERROR:ASSIGN_IN_IF found with checkpatch file check.

Signed-off-by: Carolyn Wyborny <carolyn.wyborny@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 81ad807b2601e51a6b9c0a7e4e2b25fad868ae0b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index 73edac5..4807573 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -4056,7 +4056,8 @@ static void igb_check_wvbr(struct igb_adapter *adapter)
  switch (hw->mac.type) {
  case e1000_82576:
  case e1000_i350:
-  if (!(wvbr = rd32(E1000_WVBR)))
+  wvbr = rd32(E1000_WVBR);
+  if (!wvbr)
    return;
   break;
  default:
-- 
1.7.1